mesonbuild/backend/nonebackend.py

```diff
@@ -14,6 +14,10 @@
 
 from __future__ import annotations
 
+import typing as T
+
+from mesonbuild.utils.core import MesonException
+
 from .backends import Backend
 from .. import mlog
 from ..mesonlib import MesonBugException
```

Comment:
You should import MesonException from mesonlib, not utils.core (the latter is primarily to allow having only parts of mesonlib imported during extremely speed-critical subprocesses).

Reply:
Although probably these could all be mesonbugexception, since it indicates a coding error in meson itself if these arguments are ever used for the wrong backend.

Reply:
Done.  See [c3221b2](https://github.com/mesonbuild/meson/pull/11049/commits/c3221b2ef161c2247c55e1bbe653f25fe5e58c1e)